### PR TITLE
fix(lineage): Include maxHops in Lineage Cache Key + misc UI improvements

### PR DESCRIPTION
**Pull request by jjoyce0510**
## Summary

Previously we changed the computation of the lineage cache key to omit the maxHops value which produces incorrectly cached results when the lineage cache is enabled. Note that this ONLY impacts folks who have explicitly set `LINEAGE_SEARCH_CACHE_ENABLED` to true in gms environment variable.

Changes

1. Add maxHops to cache key for lineage + Add Tests to prevent regression. 
2. Enable cursor on mouseover of Lineage Timeline view 
3. Default to All Time in lineage views to align with what currently happens when you click the lineage button 

## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: datahub-web-react/src/app/lineage/LineageTimeSelector.tsx [hunk 1]
```diff
@@ -12,6 +12,7 @@ const RangePickerWrapper = styled.div`
     position: relative;
     &:hover {
         background-color: ${ANTD_GRAY[2]};
+        cursor: pointer;
```
Comment (aditya-radhakrishnan) [inline datahub-web-react/src/app/lineage/LineageTimeSelector.tsx:15]: Nice!

File: metadata-io/src/main/java/com/linkedin/metadata/search/EntityLineageResultCacheKey.java [hunk 2]
```diff
@@ -11,4 +11,5 @@ public class EntityLineageResultCacheKey {
   private final LineageDirection direction;
   private final Long startTimeMillis;
   private final Long endTimeMillis;
+  private final Integer maxHops;
```
Comment (aditya-radhakrishnan) [inline metadata-io/src/main/java/com/linkedin/metadata/search/EntityLineageResultCacheKey.java:14]: Thanks for adding this

**Review by chriscollins3456 — APPROVED** [review]
this looks good to me! I'm definitely pro not making last 14 days the default (especially for now)

File: datahub-web-react/src/app/entity/shared/containers/profile/EntityProfile.tsx [hunk 3]
```diff
@@ -192,22 +190,14 @@ export const EntityProfile = <T, U>({
             tabParams?: Record<string, any>;
             method?: 'push' | 'replace';
         }) => {
-            let modifiedTabParams = tabParams;
-            if (tabName === 'Lineage') {
-                modifiedTabParams = {
-                    ...tabParams,
-                    start_time_millis: dayjs().subtract(14, 'day').valueOf(),
```
Comment (aditya-radhakrishnan) [inline datahub-web-react/src/app/entity/shared/containers/profile/EntityProfile.tsx:199]: Why are we removing this? Does this mean we show All Time by default?

File: metadata-io/src/test/java/com/linkedin/metadata/search/LineageSearchServiceTest.java [hunk 4]
```diff
@@ -233,6 +236,59 @@ public void testSearchService() throws Exception {
     assertEquals(searchResult.getEntities().size(), 0);
     clearCache();
 
+    // Test Cache Behavior
+    Mockito.reset(_graphService);
+
+    // Case 1: Use the maxHops in the cache.
```
Comment (aditya-radhakrishnan) [inline metadata-io/src/test/java/com/linkedin/metadata/search/LineageSearchServiceTest.java:242]: Thanks for adding a test here

**Review by aditya-radhakrishnan — APPROVED** [review]
One minor question

**Pull request merged**